### PR TITLE
[WIP] Allow Wrapper Queries

### PR DESCRIPTION
**Pull request by alexanderkiel**
It's work in progress open for discussions. I don't like the `:wrapper-class` thing but don't know how to get the query from the factory. Also don't tested any remote stuff.

### CODE REVIEW

File: src/main/compassus/core.cljc [hunk 1]
```diff
@@ -17,7 +17,7 @@
   (-> app :config :root-class))
 
 (defn- make-root-class
-  [{:keys [routes wrapper history]}]
+  [{:keys [routes wrapper wrapper-class history]}]
```
Comment (anmonteiro) [inline src/main/compassus/core.cljc]: we can probably get at the query by invoking `om.next/react-type` on the wrapper thing and checking if it implements a query with `om.next/iquery?`

**Comment by anmonteiro** [comment]
I made a comment on the source.
I'd also like to see some tests for this, and possibly a devcards example

File: src/main/compassus/core.cljc (hunk 1 above)
Comment (alexanderkiel) [inline src/main/compassus/core.cljc]: wrapper is the component factory. as I understand it, a factory is really only a function. `(om/iquery? (om/react-type wrapper))` returns false.

**Comment by alexanderkiel** [comment]
Regarding tests and devcard example: yes for sure after the implementation makes sense.

**Comment by anmonteiro** [comment]
let's do it like this:

``` clojure
(defn ^boolean is-factory? [wrapper]
  (= (subs (pr-str wrapper) 0 20)
     "#object[om$next$self"))

wrapper-query (when (is-factory? wrapper)
                        (let [c (wrapper)
                              class (om/react-type c)]
                          (when (om/iquery? class)
                            (om/get-query class))))
```

this way there's no need to introduce a `:wrapper-class` option, which is definitely undesirable.

Still need to check if this works with advanced optimizations though

**Comment by alexanderkiel** [comment]
What do you think about my commit d0824b8? I saw the problem with `om/full-query` while working with remotes. Is my idea to rename everything into static... a good one?

**Comment by anmonteiro** [comment]
this has been addressedn in b63876a806bb03b3e462754136bfcd24fdf2001b.
See #3 for an explanation.

**Pull request closed without merging**